### DISCUSSION TITLE
Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out

### DISCUSSION THREAD

**Discussion by pankajthamkeappdirect** [Netflix/conductor#2941] [Q&A] [answered]
When I execute the workflow, most of the task gets completed but for few tasks, I am facing below issue. Please suggest.

Failed to invoke HTTP task due to: java.lang.Exception: I/O error on POST request for "https://26.example.com/api/notifications/v1/WORKFLOW_REPORT": Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out

Workflow: 
{
        "name": "send_notification",
        "taskReferenceName": "notify_workflow_consumer_workflow_failure",
        "inputParameters": {
          "http_request": {
            "uri": "${workflow.BaseURL}/api/notifications/v1/WORKFLOW_REPORT",
            "method": "POST",
            "accept": "application/json",
            "body": {
              "recipients": [
                {
                  "type": "",
                  "identifier": ""
                }
              ],
              "variables": {
                "WORKFLOW_NAME": "",
                "WORKFLOW_ID": "",
                "WORKFLOW_ENTITY_ID": "",
                "WORKFLOW_ENTITY_TYPE": "",
                "WORKFLOW_ENTITY_NAME": "",
                "WORKFLOW_RESULT": "error",
                "WORKFLOW_REPORT": ""
              }
            }
          }
        },
        "type": "HTTP",
        "startDelay": 0,
        "optional": true
      }

**Comment by v1r3n** [comment, marked as answer]
You can increase the HTTP read timeouts in the request.  
e.g. the configuration below (readTimeout after accept parameter) will set the readTimeout for the request to 12 seconds.

```
{
   "name":"send_notification",
   "taskReferenceName":"notify_workflow_consumer_workflow_failure",
   "inputParameters":{
      "http_request":{
         "uri":"${workflow.BaseURL}/api/notifications/v1/WORKFLOW_REPORT",
         "method":"POST",
         "accept":"application/json",
         "readTimeOut": 12000,
         "body":{
            "recipients":[
               {
                  "type":"",
                  "identifier":""
               }
            ],
            "variables":{
               "WORKFLOW_NAME":"",
               "WORKFLOW_ID":"",
               "WORKFLOW_ENTITY_ID":"",
               "WORKFLOW_ENTITY_TYPE":"",
               "WORKFLOW_ENTITY_NAME":"",
               "WORKFLOW_RESULT":"error",
               "WORKFLOW_REPORT":""
            }
         }
      }
   },
   "type":"HTTP",
   "startDelay":0,
   "optional":true
}
```

**Reply by pankajthamkeappdirect** [reply]
Thanks, it is working for me.
